src/emm/services/modules_service.py

```diff
@@ -135,3 +148,73 @@ def sync_module(self, module_name: str, module_data: EvgModule) -> None:
 
         self.git_service.fetch(module_location)
         self.git_service.checkout(module_revision, directory=module_location)
+
+    def attempt_git_operation(
+        self,
+        operation: GitAction,
+        revision: str,
+        branch_name: Optional[str] = None,
+        directory: Optional[Path] = None,
+    ) -> Optional[str]:
+        """
+        Attempt to perform the specified git operation.
+
+        :param operation: Git operation to perform.
+        :param revision: Git revision to perform operation on.
+        :param branch_name: Name of branch for git checkout.
+        :param directory: Directory of git repository.
+        :return: Error message if an error was encountered.
+        """
+        try:
+            self.git_service.perform_git_action(operation, revision, branch_name, directory)
+        except ProcessExecutionError:
+            LOGGER.warning("Error encountered during git operation", exc_info=True)
+            return f"Encountered error performing '{operation}' on '{revision}'"
+        return None
+
+    def git_operate_modules(
+        self, operation: GitAction, branch: str, enabled_modules: Dict[str, EvgModule]
+    ) -> Dict[str, str]:
+        """
+        Git operate modules to the specific revisions.
+
+        :param operation: Git operation to perform.
+        :param branch: Name of branch for git checkout.
+        :param enabled_modules: Dictionary of enabled modules.
+        :return: Dictionary of error encountered.
+        """
+        error_encountered = {}
+        manifest = self.get_evg_manifest(self.emm_options.evg_project)
+        manifest_modules = manifest.modules
+        if manifest_modules is None:
+            raise ValueError("Modules not found in manifest")
```

Comment:
[nit] It might be useful to include some vertical whitespace in the function to make the code easier to read. For example, adding a blank line after raising exceptions can help visually break up the function into meaningful chunks.